### PR TITLE
HDDS-8718. Kerberos authentication does not work for GrpcOmTransport

### PR DESCRIPTION
**Pull request by ivanzlenko**
## What changes were proposed in this pull request?

Fixed issue where it is not possible to retrieve UGI for getSecret and revokeSecret requests while OM communicating with client using GPRC transport. 

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-8718

## How was this patch tested?

Tested with integration tests introduced in pull request for HDDS-8050

### CODE REVIEW

File: hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/s3/security/S3SecretRequestHelperTest.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import org.apache.hadoop.ipc.Server;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.junit.jupiter.api.Test;
+
+import static org.apache.hadoop.security.SaslRpcServer.AuthMethod.KERBEROS;
+import static org.apache.hadoop.security.UserGroupInformation.createRemoteUser;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+class S3SecretRequestHelperTest {
```
Comment (adoroszlai) [inline hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/s3/security/S3SecretRequestHelperTest.java:31]: Please rename to `TestS3SecretRequestHelper`.  Only tests that follow naming convention `Test...` are run by Ozone's `mvn test`.

**Review by adoroszlai — CHANGES_REQUESTED** [review]
Thanks @ivanzlenko for the patch.

**Comment by adoroszlai** [comment]
@ashishkumar50 @Tejaskriya please take a look

File: hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/s3/security/S3SecretRequestHelperTest.java (hunk 1 above)
Comment (ivanzlenko) [reply hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/s3/security/S3SecretRequestHelperTest.java]: Thanks! Renamed.

**Comment by adoroszlai** [comment]
Thanks @ivanzlenko for updating the patch.  The new test is failing with:

```
java.lang.NoClassDefFoundError: Could not initialize class org.apache.hadoop.ozone.om.request.s3.security.TestS3SecretRequestHelper
...
Caused by: java.lang.NullPointerException
	at org.apache.hadoop.security.authentication.util.KerberosName.getShortName(KerberosName.java:422)
	at org.apache.hadoop.security.User.<init>(User.java:48)
	at org.apache.hadoop.security.User.<init>(User.java:43)
	at org.apache.hadoop.security.UserGroupInformation.createRemoteUser(UserGroupInformation.java:1426)
	at org.apache.hadoop.ozone.om.request.s3.security.TestS3SecretRequestHelper.<clinit>(TestS3SecretRequestHelper.java:34)
```

https://github.com/ivanzlenko/ozone/actions/runs/5207802223/jobs/9395707878#step:6:2408

File: hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/s3/security/TestS3SecretRequestHelper.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      super(null);
+      this.ugi = ugi;
+    }
+
+    @Override
+    public UserGroupInformation getRemoteUser() {
+      return ugi;
+    }
+  }
+}
```
Comment (ashishkumar50) [inline hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/s3/security/TestS3SecretRequestHelper.java:67]: Add new line at the end of file.

**Review by ashishkumar50 — COMMENTED** [review]
Hi @ivanzlenko, Thanks for working on this. Whether `getOrCreateUgi(accessId)` required even for S3SetSecret request as well?

File: hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/s3/security/TestS3SecretRequestHelper.java (hunk 2 above)
Comment (ivanzlenko) [reply hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/s3/security/TestS3SecretRequestHelper.java:67]: @ashishkumar50 yeah, it make sense. Thanks for pointing.

Comment (ivanzlenko) [reply hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/s3/security/TestS3SecretRequestHelper.java:92]: @ashishkumar50 I can see new line in IDE though it's omitted in GitHub. Fixed unit tests and set s3 secret request.

**Review by adoroszlai — COMMENTED** [review]
Thanks @ivanzlenko for the patch, it works fine (tested in `ozonesecure` env. TLS turned off).

**Comment by ivanzlenko** [comment]
@kerneltime @duongkame @neils-dev can you please take a look at this review?

**Review by adoroszlai — APPROVED** [review]

**Comment by ivanzlenko** [comment]
Thanks @adoroszlai !

**Pull request merged**